File: app/models/OrgConfig.java
package models;

import java.io.IOException;
import java.text.SimpleDateFormat;
import java.util.HashMap;
import java.util.TimeZone;

public class OrgConfig {
    public String name;
    public String people_url;
    public String str_manual_title;
    public String str_manual_title_short;
    public String str_res_plan_short;
    public String str_res_plan;
    public String str_res_plan_cap;
    public String str_res_plans;
    public String str_res_plans_cap;
    public String str_jc_name = "Judicial Committee";
    public String str_jc_name_short = "JC";
    public String str_findings = "Findings";
    public String str_corporation = "Corporation";
    public String str_committee = "Committee";
    public String str_clerk = "Clerk";
    public String str_guilty = "Guilty";
    public String str_not_guilty = "Not Guilty";

    public boolean show_no_contest_plea = false;
    public boolean show_na_plea = false;
    public boolean show_severity = false;
    public boolean use_minor_referrals = true;
    public boolean show_checkbox_for_res_plan = true;
    public boolean track_writer = true;
    public boolean filter_no_charge_cases = false;
    public boolean show_findings_in_rp_list = true;
    public boolean use_year_in_case_number = false;
    public boolean hide_location_for_print = false;
    public boolean euro_dates = false;

    public boolean enable_file_sharing = false;

    public Organization org;

    public TimeZone time_zone = TimeZone.getTimeZone("US/Eastern");

    public static OrgConfig get() {
        Organization org = Organization.getByHost();
        OrgConfig result = configs.get(org.name);
        result.org = org;
        return result;
    }

    public String getReferralDestination(Charge c) {
        return "School Meeting";
    }

    static HashMap<String, OrgConfig> configs = new HashMap<String, OrgConfig>();
    static void register(String name, OrgConfig config) {
        configs.put(name, config);
    }

    public String getCaseNumberPrefix(Meeting m) {
        String format = null;

        if (use_year_in_case_number) {
            format = euro_dates ? "dd-MM-YYYY-" : "YYYY-MM-dd-";
        } else {
            format = euro_dates ? "dd-MM-" : "MM-dd-";
        }
        return new SimpleDateFormat(format).format(m.date);
    }

    public String translatePlea(String plea) {
        if (plea.equals("Guilty")) {
            return this.str_guilty;
        } else if (plea.equals("Not Guilty")) {
            return this.str_not_guilty;
        }

        return plea;
    }
}

class ThreeRiversVillageSchool extends OrgConfig {
    private static final ThreeRiversVillageSchool INSTANCE = new ThreeRiversVillageSchool();

    public ThreeRiversVillageSchool() {
        name = "Three Rivers Village School";
        people_url = "https://trvs.demschooltools.com";

        str_manual_title = "Management Manual";
        str_manual_title_short = "Manual";
        str_res_plan_short = "RP";
        str_res_plan = "resolution plan";
        str_res_plan_cap = "Resolution plan";
        str_res_plans = "resolution plans";
        str_res_plans_cap = "Resolution plans";
        str_jc_name = "Justice Committee";

        show_checkbox_for_res_plan = false;

        enable_file_sharing = true;

        OrgConfig.register(name, this);
    }

    public static ThreeRiversVillageSchool getInstance() {
        return INSTANCE;
    }
}

class PhillyFreeSchool extends OrgConfig {
    private static final PhillyFreeSchool INSTANCE = new PhillyFreeSchool();

    public PhillyFreeSchool() {
        name = "Philly Free School";
        people_url = "https://pfs.demschooltools.com";

        str_manual_title = "Lawbook";
        str_manual_title_short = "Lawbook";
        str_res_plan_short = "Sentence";
        str_res_plan = "sentence";
        str_res_plan_cap = "Sentence";
        str_res_plans = "sentences";
        str_res_plans_cap = "Sentences";

        show_no_contest_plea = true;
        show_na_plea = true;
        show_severity = true;

        OrgConfig.register(name, this);
    }

    public static PhillyFreeSchool getInstance() {
        return INSTANCE;
    }
}

class Fairhaven extends OrgConfig {
    private static final Fairhaven INSTANCE = new Fairhaven();

    public Fairhaven() {
        name = "Fairhaven School";
        people_url = "https://fairhaven.demschooltools.com";

        str_manual_title = "Lawbook";
        str_manual_title_short = "Lawbook";
        str_res_plan_short = "Sentence";
        str_res_plan = "sentence";
        str_res_plan_cap = "Sentence";
        str_res_plans = "sentences";
        str_res_plans_cap = "Sentences";
		str_findings = "JC Report";

        show_findings_in_rp_list = false;
        use_year_in_case_number = true;

        OrgConfig.register(name, this);
    }

    public static Fairhaven getInstance() {
        return INSTANCE;
    }
}

class TheCircleSchool extends OrgConfig {
    private static final TheCircleSchool INSTANCE = new TheCircleSchool();

    public TheCircleSchool() {
        name = "The Circle School";
        people_url = "https://tcs.demschooltools.com";

        str_manual_title = "Management Manual";
        str_manual_title_short = "Mgmt. Man.";
        str_res_plan_short = "Sentence";
        str_res_plan = "sentence";
        str_res_plan_cap = "Sentence";
        str_res_plans = "sentences";
        str_res_plans_cap = "Sentences";
		str_findings = "Findings";

        track_writer = false;
        filter_no_charge_cases = true;
        hide_location_for_print = true;

        OrgConfig.register(name, this);
    }

    @Override
    public String getReferralDestination(Charge c) {
        if (c.plea.equals("Not Guilty")) {
            return "trial";
        } else {
            return "School Meeting";
        }
    }

    public static TheCircleSchool getInstance() {
        return INSTANCE;
    }

    @Override
    public String getCaseNumberPrefix(Meeting m) {
        return new SimpleDateFormat("yyyyMMdd").format(m.date);
    }
}

class MakariosLearningCommunity extends OrgConfig {
    private static final MakariosLearningCommunity INSTANCE =
        new MakariosLearningCommunity();

    public MakariosLearningCommunity() {
        name = "Makarios Learning Community";
        people_url = "https://mlc.demschooltools.com";
        time_zone = TimeZone.getTimeZone("US/Central");

        str_manual_title = "Management Manual";
        str_manual_title_short = "Mgmt. Man.";
        str_res_plan_short = "Sentence";
        str_res_plan = "sentence";
        str_res_plan_cap = "Sentence";
        str_res_plans = "sentences";
        str_res_plans_cap = "Sentences";
        str_findings = "Findings";

        OrgConfig.register(name, this);
    }

    public static MakariosLearningCommunity getInstance() {
        return INSTANCE;
    }
}

class TheOpenSchool extends OrgConfig {
    private static final TheOpenSchool INSTANCE = new TheOpenSchool();

    public TheOpenSchool() {
        name = "The Open School";
        people_url = "https://tos.demschooltools.com";
        time_zone = TimeZone.getTimeZone("US/Pacific");

        str_manual_title = "Law Book";
        str_manual_title_short = "Law Book";
        str_res_plan_short = "Sentence";
        str_res_plan = "sentence";
        str_res_plan_cap = "Sentence";
        str_res_plans = "sentences";
        str_res_plans_cap = "Sentences";
        str_findings = "Findings";
        str_jc_name = "Civics Board";
        str_jc_name_short = "CB";
        str_guilty = "Agree";
        str_not_guilty = "Disagree";

        show_findings_in_rp_list = true;
        use_minor_referrals = false;
        show_no_contest_plea = true;

        OrgConfig.register(name, this);
    }

    public static TheOpenSchool getInstance() {
        return INSTANCE;
    }
}

class TheOpenSchool2 extends OrgConfig {
    private static final TheOpenSchool2 INSTANCE = new TheOpenSchool2();

    public TheOpenSchool2() {
        name = "The Open School - Riverside";
        people_url = "https://tos-2.demschooltools.com";
        time_zone = TimeZone.getTimeZone("US/Pacific");

        str_manual_title = "Law Book";
        str_manual_title_short = "Law Book";
        str_res_plan_short = "Sentence";
        str_res_plan = "sentence";
        str_res_plan_cap = "Sentence";
        str_res_plans = "sentences";
        str_res_plans_cap = "Sentences";
        str_findings = "Findings";
        str_jc_name = "Justice Committee";
        str_jc_name_short = "JC";
        str_guilty = "Confirm";
        str_not_guilty = "Deny";

        show_findings_in_rp_list = true;
        use_minor_referrals = false;
        show_no_contest_plea = true;

        OrgConfig.register(name, this);
    }

    public static TheOpenSchool2 getInstance() {
        return INSTANCE;
    }
}

class Houston extends OrgConfig {
    private static final Houston INSTANCE = new Houston();
    public TimeZone time_zone = TimeZone.getTimeZone("US/Central");

    public Houston() {
        name = "Houston Sudbury School";
        people_url = "https://hss.demschooltools.com";

        str_manual_title = "Lawbook";
        str_manual_title_short = "Lawbook";
        str_res_plan_short = "Sentence";
        str_res_plan = "sentence";
        str_res_plan_cap = "Sentence";
        str_res_plans = "sentences";
        str_res_plans_cap = "Sentences";
        str_findings = "Findings";

        track_writer = true;
        use_year_in_case_number = true;

        OrgConfig.register(name, this);
    }

    public static Houston getInstance() {
        return INSTANCE;
    }
}


class Sandbox extends OrgConfig {
    private static final Sandbox INSTANCE = new Sandbox();
    public TimeZone time_zone = TimeZone.getTimeZone("US/Eastern");

    public Sandbox() {
        name = "DemSchoolTools sandbox area";
        people_url = "https://sandbox.demschooltools.com";

        str_manual_title = "Lawbook";
        str_manual_title_short = "Lawbook";
        str_res_plan_short = "Sentence";
        str_res_plan = "sentence";
        str_res_plan_cap = "Sentence";
        str_res_plans = "sentences";
        str_res_plans_cap = "Sentences";
        str_findings = "Findings";

        track_writer = true;
        enable_file_sharing = true;

        OrgConfig.register(name, this);
    }

    public static Sandbox getInstance() {
        return INSTANCE;
    }
}

class Clearview extends OrgConfig {
    private static final Clearview INSTANCE = new Clearview();
    public TimeZone time_zone = TimeZone.getTimeZone("US/Central");

    public Clearview() {
        name = "Clearview Sudbury School";
        people_url = "https://css.demschooltools.com";

        str_manual_title = "Lawbook";
        str_manual_title_short = "Lawbook";
        str_res_plan_short = "Sentence";
        str_res_plan = "sentence";
        str_res_plan_cap = "Sentence";
        str_res_plans = "sentences";
        str_res_plans_cap = "Sentences";
        str_findings = "Findings";

        track_writer = false;

        OrgConfig.register(name, this);
    }

    public static Clearview getInstance() {
        return INSTANCE;
    }
}

class Wicklow extends OrgConfig {
    private static final Wicklow INSTANCE = new Wicklow();
    public TimeZone time_zone = TimeZone.getTimeZone("Europe/Dublin");

    public Wicklow() {
        name = "Wicklow Democratic School";
        people_url = "https://wicklow.demschooltools.com";

        str_manual_title = "Lawbook";
        str_manual_title_short = "Lawbook";
        str_res_plan_short = "Sanction";
        str_res_plan = "sanction";
        str_res_plan_cap = "Sanction";
        str_res_plans = "sanctions";
        str_res_plans_cap = "Sanctions";
        str_findings = "Findings";

        track_writer = false;
        euro_dates = true;
        use_year_in_case_number = true;

        OrgConfig.register(name, this);
    }

    public static Wicklow getInstance() {
        return INSTANCE;
    }
}

class Tallgrass extends OrgConfig {
    private static final Tallgrass INSTANCE = new Tallgrass();
    public TimeZone time_zone = TimeZone.getTimeZone("US/Central");

    public Tallgrass() {
        name = "Tallgrass Sudbury School";
        people_url = "https://tallgrass.demschooltools.com";

        str_manual_title = "Lawbook";
        str_manual_title_short = "Lawbook";
        str_res_plan_short = "Result";
        str_res_plan = "result";
        str_res_plan_cap = "Result";
        str_res_plans = "results";
        str_res_plans_cap = "Results";
        str_findings = "Findings";
        str_jc_name = "Restoration Committee";
        str_jc_name_short = "RC";

        str_guilty = "Agree";
        str_not_guilty = "Disagree";

        track_writer = false;

        OrgConfig.register(name, this);
    }

    public static Tallgrass getInstance() {
        return INSTANCE;
    }
}

class MiamiSudburySchool extends OrgConfig {
    private static final MiamiSudburySchool INSTANCE = new MiamiSudburySchool();
    public TimeZone time_zone = TimeZone.getTimeZone("US/Eastern");

    public MiamiSudburySchool() {
        name = "Miami Sudbury School";
        people_url = "https://miami.demschooltools.com";

        str_manual_title = "Lawbook";
        str_manual_title_short = "Lawbook";
        str_res_plan_short = "Ruling";
        str_res_plan = "ruling";
        str_res_plan_cap = "Ruling";
        str_res_plans = "rulings";
        str_res_plans_cap = "Rulings";
        str_findings = "Complaint & Further Information";

        str_guilty = "Accepts responsibility";
        str_not_guilty = "Does not accept responsibility";

        track_writer = true;

        OrgConfig.register(name, this);
    }

    public static MiamiSudburySchool getInstance() {
        return INSTANCE;
    }
}

class SligoSudburySchool extends OrgConfig {
    private static final SligoSudburySchool INSTANCE = new SligoSudburySchool();
    public TimeZone time_zone = TimeZone.getTimeZone("Europe/Dublin");

    public SligoSudburySchool() {
        name = "Sligo Sudbury School";
        people_url = "https://sligo.demschooltools.com";

        str_manual_title = "Lawbook";
        str_manual_title_short = "Lawbook";
        str_res_plan_short = "Sanction";
        str_res_plan = "sanction";
        str_res_plan_cap = "Sanction";
        str_res_plans = "sanctions";
        str_res_plans_cap = "Sanctions";

        track_writer = true;
        euro_dates = true;
        use_year_in_case_number = true;

        OrgConfig.register(name, this);
    }

    public static SligoSudburySchool getInstance() {
        return INSTANCE;
    }
}

class LearningProjectIbiza extends OrgConfig {
    private static final LearningProjectIbiza INSTANCE = new LearningProjectIbiza();
    public TimeZone time_zone = TimeZone.getTimeZone("Europe/Madrid");

    public LearningProjectIbiza() {
        name = "Learning Project Ibiza";
        people_url = "https://tlp.demschooltools.com";

        str_manual_title = "Lawbook";
        str_manual_title_short = "Lawbook";
        str_res_plan_short = "Sentence";
        str_res_plan = "sentence";
        str_res_plan_cap = "Sentence";
        str_res_plans = "sentences";
        str_res_plans_cap = "Sentences";

        track_writer = true;
        use_year_in_case_number = true;

        OrgConfig.register(name, this);
    }

    public static LearningProjectIbiza getInstance() {
        return INSTANCE;
    }
}

class Wilmington extends OrgConfig {
    private static final Wilmington INSTANCE = new Wilmington();

    public Wilmington() {
        name = "Wilmington Sudbury School";
        people_url = "https://wilmington.demschooltools.com";

        str_manual_title = "Lawbook";
        str_manual_title_short = "Lawbook";
        str_res_plan_short = "Sentence";
        str_res_plan = "sentence";
        str_res_plan_cap = "Sentence";
        str_res_plans = "sentences";
        str_res_plans_cap = "Sentences";

        track_writer = true;
        use_year_in_case_number = true;

        OrgConfig.register(name, this);
    }

    public static Wilmington getInstance() {
        return INSTANCE;
    }
}

File: django/dst/org_config.py
import datetime
from zoneinfo import ZoneInfo

from dst.models import Organization

ALL_ORG_CONFIGS: dict[str, "OrgConfig"] = {}


def get_org_config(org: Organization) -> "OrgConfig":
    return ALL_ORG_CONFIGS[org.name]


class OrgConfig:
    def __init__(self, name):
        ALL_ORG_CONFIGS[name] = self
        self.org = Organization.objects.filter(name=name).first()

        self.name = name
        self.people_url = ""
        self.str_manual_title = ""
        self.str_manual_title_short = ""
        self.str_res_plan_short = ""
        self.str_res_plan = ""
        self.str_res_plan_cap = ""
        self.str_res_plans = ""
        self.str_res_plans_cap = ""
        self.str_jc_name = "Judicial Committee"
        self.str_jc_name_short = "JC"
        self.str_findings = "Findings"
        self.str_guilty = "Guilty"
        self.str_not_guilty = "Not Guilty"
        self.str_na = "N/A"

        self.show_no_contest_plea = False
        self.show_na_plea = False
        self.show_severity = False
        self.show_entry = True
        self.show_plea = True

        self.use_minor_referrals = True
        self.show_checkbox_for_res_plan = True
        self.track_writer = True
        self.filter_no_charge_cases = False
        self.show_findings_in_rp_list = True
        self.use_year_in_case_number = False
        self.hide_location_for_print = False
        self.euro_dates = False

        self.enable_file_sharing = False

        # TODO: Delete this and all overrides of it in subclasses of OrgConfig.
        # As of the Custodia -> DST merger, this value is now stored in the
        # Organization model. Be sure that the database is up to date with
        # these values.
        self.time_zone = ZoneInfo("US/Eastern")

    def get_referral_destination(self, charge) -> str:
        return "School Meeting"

    def get_case_number_prefix(self, meeting):
        date_format = ""

        if self.use_year_in_case_number:
            date_format = "dd-MM-YYYY-" if self.euro_dates else "YYYY-MM-dd-"
        else:
            date_format = "dd-MM-" if self.euro_dates else "MM-dd-"

        # Convert Java's SimpleDateFormat to Python's strftime format
        date_format = (
            date_format.replace("YYYY", "%Y").replace("MM", "%m").replace("dd", "%d")
        )

        return datetime.datetime.strftime(meeting.get_date(), date_format)

    def translate_plea(self, plea):
        if plea == "Guilty":
            return self.str_guilty
        elif plea == "Not Guilty":
            return self.str_not_guilty
        elif plea == "N/A":
            return self.str_na

        return plea


class ThreeRiversVillageSchool(OrgConfig):
    def __init__(self):
        super().__init__("Three Rivers Village School")
        self.people_url = "https://trvs.demschooltools.com"

        self.str_manual_title = "Management Manual"
        self.str_manual_title_short = "Manual"
        self.str_res_plan_short = "agreement"
        self.str_res_plan = "agreement"
        self.str_res_plan_cap = "Agreement"
        self.str_res_plans = "agreements"
        self.str_res_plans_cap = "Agreements"
        self.str_jc_name = "Resolution Committee"
        self.str_jc_name_short = "RC"
        self.str_findings = "Discussion Record"

        self.show_checkbox_for_res_plan = False
        self.enable_file_sharing = True
        self.use_minor_referrals = False
        self.show_entry = False
        self.show_plea = False


class PhillyFreeSchool(OrgConfig):
    def __init__(self):
        super().__init__("Philly Free School")
        self.people_url = "https://pfs.demschooltools.com"

        self.str_manual_title = "Lawbook"
        self.str_manual_title_short = "Lawbook"
        self.str_res_plan_short = "Consequence"
        self.str_res_plan = "consequence"
        self.str_res_plan_cap = "Consequence"
        self.str_res_plans = "consequences"
        self.str_res_plans_cap = "Consequences"
        self.str_guilty = "Accept Responsibility"
        self.str_not_guilty = "Reject Responsibility"

        self.show_no_contest_plea = True
        self.show_na_plea = True
        self.show_severity = True


class Fairhaven(OrgConfig):
    def __init__(self):
        super().__init__("Fairhaven School")
        self.people_url = "https://fairhaven.demschooltools.com"

        self.str_manual_title = "Lawbook"
        self.str_manual_title_short = "Lawbook"
        self.str_res_plan_short = "Sentence"
        self.str_res_plan = "sentence"
        self.str_res_plan_cap = "Sentence"
        self.str_res_plans = "sentences"
        self.str_res_plans_cap = "Sentences"
        self.str_findings = "JC Report"

        self.show_findings_in_rp_list = False
        self.use_year_in_case_number = True


class TheCircleSchool(OrgConfig):
    def __init__(self):
        super().__init__("The Circle School")
        self.people_url = "https://tcs.demschooltools.com"

        self.str_manual_title = "Management Manual"
        self.str_manual_title_short = "Mgmt. Man."
        self.str_res_plan_short = "Sentence"
        self.str_res_plan = "sentence"
        self.str_res_plan_cap = "Sentence"
        self.str_res_plans = "sentences"
        self.str_res_plans_cap = "Sentences"
        self.str_findings = "Findings"

        self.track_writer = False
        self.filter_no_charge_cases = True
        self.hide_location_for_print = True
        self.show_no_contest_plea = True

    def get_referral_destination(self, charge):
        if charge.get_plea() == "Not Guilty":
            return "trial"
        else:
            return "School Meeting"

    def get_case_number_prefix(self, meeting):
        return datetime.datetime.strftime(meeting.get_date(), "%Y%m%d")


class MakariosLearningCommunity(OrgConfig):
    def __init__(self):
        super().__init__("Makarios Learning Community")
        self.people_url = "https://mlc.demschooltools.com"
        self.time_zone = ZoneInfo("US/Central")

        self.str_manual_title = "Management Manual"
        self.str_manual_title_short = "Mgmt. Man."
        self.str_res_plan_short = "Sentence"
        self.str_res_plan = "sentence"
        self.str_res_plan_cap = "Sentence"
        self.str_res_plans = "sentences"
        self.str_res_plans_cap = "Sentences"
        self.str_findings = "Findings"


class TheOpenSchool(OrgConfig):
    def __init__(self):
        super().__init__("The Open School")
        self.people_url = "https://tos.demschooltools.com"
        self.time_zone = ZoneInfo("US/Pacific")

        self.str_manual_title = "Law Book"
        self.str_manual_title_short = "Law Book"
        self.str_res_plan_short = "Sentence"
        self.str_res_plan = "sentence"
        self.str_res_plan_cap = "Sentence"
        self.str_res_plans = "sentences"
        self.str_res_plans_cap = "Sentences"
        self.str_findings = "Findings"
        self.str_jc_name = "Civics Board"
        self.str_jc_name_short = "CB"
        self.str_guilty = "Agree"
        self.str_not_guilty = "Disagree"
        self.str_na = "Mediated"

        self.show_findings_in_rp_list = True
        self.use_minor_referrals = False
        self.show_no_contest_plea = True
        self.show_na_plea = True


class TheOpenSchoolVirtual(OrgConfig):
    def __init__(self):
        super().__init__("The Open School Virtual")
        self.people_url = "https://tosv.demschooltools.com"
        self.time_zone = ZoneInfo("US/Pacific")

        self.str_manual_title = "Law Book"
        self.str_manual_title_short = "Law Book"
        self.str_res_plan_short = "Sentence"
        self.str_res_plan = "sentence"
        self.str_res_plan_cap = "Sentence"
        self.str_res_plans = "sentences"
        self.str_res_plans_cap = "Sentences"
        self.str_findings = "Findings"
        self.str_jc_name = "Civics Board"
        self.str_jc_name_short = "CB"
        self.str_guilty = "Agree"
        self.str_not_guilty = "Disagree"
        self.str_na = "Mediated"

        self.show_findings_in_rp_list = True
        self.use_minor_referrals = False
        self.show_no_contest_plea = True
        self.show_na_plea = True


class Houston(OrgConfig):
    def __init__(self):
        super().__init__("Houston Sudbury School")
        self.people_url = "https://hss.demschooltools.com"
        self.time_zone = ZoneInfo("US/Central")

        self.str_manual_title = "Lawbook"
        self.str_manual_title_short = "Lawbook"
        self.str_res_plan_short = "Sentence"
        self.str_res_plan = "sentence"
        self.str_res_plan_cap = "Sentence"
        self.str_res_plans = "sentences"
        self.str_res_plans_cap = "Sentences"
        self.str_findings = "Findings"

        self.track_writer = True
        self.use_year_in_case_number = True


class Sandbox(OrgConfig):
    def __init__(self):
        super().__init__("DemSchoolTools sandbox area")
        self.people_url = "https://sandbox.demschooltools.com"
        self.time_zone = ZoneInfo("US/Eastern")

        self.str_manual_title = "Lawbook"
        self.str_manual_title_short = "Lawbook"
        self.str_res_plan_short = "Sentence"
        self.str_res_plan = "sentence"
        self.str_res_plan_cap = "Sentence"
        self.str_res_plans = "sentences"
        self.str_res_plans_cap = "Sentences"
        self.str_findings = "Findings"

        self.track_writer = True
        self.enable_file_sharing = True


class Clearview(OrgConfig):
    def __init__(self):
        super().__init__("Clearview Sudbury School")
        self.people_url = "https://css.demschooltools.com"
        self.time_zone = ZoneInfo("US/Central")

        self.str_manual_title = "Rulebook"
        self.str_manual_title_short = "Rulebook"
        self.str_res_plan_short = "Resolution"
        self.str_res_plan = "resolution"
        self.str_res_plan_cap = "Resolution"
        self.str_res_plans = "resolutions"
        self.str_res_plans_cap = "Resolutions"
        self.str_findings = "Findings"

        self.str_guilty = "Yes"
        self.str_not_guilty = "No"

        self.track_writer = True


class Wicklow(OrgConfig):
    def __init__(self):
        super().__init__("Wicklow Democratic School")
        self.people_url = "https://wicklow.demschooltools.com"
        self.time_zone = ZoneInfo("Europe/Dublin")

        self.str_manual_title = "Lawbook"
        self.str_manual_title_short = "Lawbook"
        self.str_res_plan_short = "Sanction"
        self.str_res_plan = "sanction"
        self.str_res_plan_cap = "Sanction"
        self.str_res_plans = "sanctions"
        self.str_res_plans_cap = "Sanctions"
        self.str_findings = "Findings"

        self.track_writer = False
        self.euro_dates = True
        self.use_year_in_case_number = True


class Tallgrass(OrgConfig):
    def __init__(self):
        super().__init__("Tallgrass Sudbury School")
        self.people_url = "https://tallgrass.demschooltools.com"
        self.time_zone = ZoneInfo("US/Central")

        self.str_manual_title = "Lawbook"
        self.str_manual_title_short = "Lawbook"
        self.str_res_plan_short = "Result"
        self.str_res_plan = "result"
        self.str_res_plan_cap = "Result"
        self.str_res_plans = "results"
        self.str_res_plans_cap = "Results"
        self.str_findings = "Findings"
        self.str_jc_name = "Restoration Committee"
        self.str_jc_name_short = "RC"

        self.str_guilty = "Agree"
        self.str_not_guilty = "Disagree"

        self.track_writer = False


class MiamiSudburySchool(OrgConfig):
    def __init__(self):
        super().__init__("Miami Sudbury School")
        self.people_url = "https://miami.demschooltools.com"
        self.time_zone = ZoneInfo("US/Eastern")

        self.str_manual_title = "Lawbook"
        self.str_manual_title_short = "Lawbook"
        self.str_res_plan_short = "Ruling"
        self.str_res_plan = "ruling"
        self.str_res_plan_cap = "Ruling"
        self.str_res_plans = "rulings"
        self.str_res_plans_cap = "Rulings"
        self.str_findings = "Complaint & Further Information"

        self.str_guilty = "Accepts responsibility"
        self.str_not_guilty = "Does not accept responsibility"

        self.track_writer = True


class SligoSudburySchool(OrgConfig):
    def __init__(self):
        super().__init__("Sligo Sudbury School")
        self.people_url = "https://sligo.demschooltools.com"
        self.time_zone = ZoneInfo("Europe/Dublin")

        self.str_manual_title = "Lawbook"
        self.str_manual_title_short = "Lawbook"
        self.str_res_plan_short = "Sanction"
        self.str_res_plan = "sanction"
        self.str_res_plan_cap = "Sanction"
        self.str_res_plans = "sanctions"
        self.str_res_plans_cap = "Sanctions"

        self.track_writer = True
        self.euro_dates = True
        self.use_year_in_case_number = True


class LearningProjectIbiza(OrgConfig):
    def __init__(self):
        super().__init__("Learning Project Ibiza")
        self.people_url = "https://tlp.demschooltools.com"
        self.time_zone = ZoneInfo("Europe/Madrid")

        self.str_manual_title = "Lawbook"
        self.str_manual_title_short = "Lawbook"
        self.str_res_plan_short = "Sentence"
        self.str_res_plan = "sentence"
        self.str_res_plan_cap = "Sentence"
        self.str_res_plans = "sentences"
        self.str_res_plans_cap = "Sentences"

        self.track_writer = True
        self.use_year_in_case_number = True


class Wilmington(OrgConfig):
    def __init__(self):
        super().__init__("Wilmington Sudbury School")
        self.people_url = "https://wilmington.demschooltools.com"

        self.str_manual_title = "Lawbook"
        self.str_manual_title_short = "Lawbook"
        self.str_res_plan_short = "Sentence"
        self.str_res_plan = "sentence"
        self.str_res_plan_cap = "Sentence"
        self.str_res_plans = "sentences"
        self.str_res_plans_cap = "Sentences"

        self.track_writer = True
        self.use_year_in_case_number = True


class SouthJersey(OrgConfig):
    def __init__(self):
        super().__init__("South Jersey Sudbury School")
        self.people_url = "https://sjss.demschooltools.com"

        self.str_manual_title = "Rulebook"
        self.str_manual_title_short = "Rulebook"
        self.str_res_plan_short = "Sentence"
        self.str_res_plan = "sentence"
        self.str_res_plan_cap = "Sentence"
        self.str_res_plans = "sentences"
        self.str_res_plans_cap = "Sentences"

        self.track_writer = True
        self.use_year_in_case_number = True


# Initialize all the configurations
# They will be automatically registered in ALL_ORG_CONFIGS
ThreeRiversVillageSchool()
PhillyFreeSchool()
Fairhaven()
TheCircleSchool()
MakariosLearningCommunity()
TheOpenSchool()
TheOpenSchoolVirtual()
Houston()
Sandbox()
Clearview()
Wicklow()
Tallgrass()
MiamiSudburySchool()
SligoSudburySchool()
LearningProjectIbiza()
Wilmington()
SouthJersey()
